seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -140,17 +143,28 @@ def test_create_project_from_workspace(self, mock_file_exist, mock_email, mock_a
         mock_file_exist.return_value = False
         response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_NO_SLASH_DATA_PATH))
         self.assertEqual(response.status_code, 400)
-        self.assertEqual(response.json()['error'], 'Data file or path test_no_slash_path is not found.')
-        mock_file_exist.assert_called_with('gs://test_bucket/test_no_slash_path', user=self.manager_user)
+        self.assertEqual(response.json()['error'], 'Data file or path test_no_slash_path.vcf.bgz is not found.')
+        mock_file_exist.assert_called_with('gs://test_bucket/test_no_slash_path.vcf.bgz', user=self.manager_user)
         mock_has_service_account.assert_called_with(self.manager_user, TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME)
         self.assertEqual(mock_has_service_account.call_count, 1)
         self.assertEqual(mock_sleep.call_count, 1)
 
+        response = self.client.post(url, content_type='application/json',
+                                    data=json.dumps(REQUEST_BODY_BAD_DATA_PATH))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json()['error'], 'The data file must have a postfix of ".vcf", ".vcf.gz", or ".vcf.bgz".')
+
+        mock_file_exist.return_value = True
+        mock_file_iter.return_value = ['##fileformat=VCFv4.2\n', '#CHROM	POS	ID	REF	ALT	QUAL']  # incomplete header line
```

Comment:
`file_iter` will return a string representation of a line in the file, not a list of strings

Reply:
I know that `file_iter()` returns an iteration of lines in the file. The list is iterable so that it can emulate the return value. Maybe adding `iter()` to the list is better but it seems not necessary.

Reply:
oh yeah good point, this is fine to keep as is. I would say add a couple of "lines" after the header in the list just to test that iteration stops correctly